### PR TITLE
Add an option for setting the threshold at which statements are considered slow

### PR DESCRIPTION
**Pull request by jbearer**

### CODE REVIEW

File: src/data_source/storage/sql.rs [hunk 1]
```diff
@@ -174,11 +175,8 @@ fn add_custom_migrations(
 
 /// Postgres client config.
 #[derive(Clone, Debug)]
-pub struct Config<DB>
-where
-    DB: Database,
-{
-    db_opt: <DB::Connection as Connection>::Options,
+pub struct Config {
+    db_opt: PgConnectOptions,
```
Comment (imabdulbasit) [inline src/data_source/storage/sql.rs:179]: So we would need two Configs when we make `Sqlite` change with that feature thing?

Comment (imabdulbasit) [reply src/data_source/storage/sql.rs:179]: Okay, I will revert this change then in Sqlite PR

File: src/data_source/storage/sql.rs [hunk 2]
```diff
@@ -364,6 +360,16 @@ impl<DB: Database> Config<DB> {
         self.pool_opt = self.pool_opt.max_connections(max);
         self
     }
+
+    /// Log at WARN level any time a SQL statement takes longer than `threshold`.
+    ///
+    /// The default threshold is 1s.
+    pub fn slow_statement_threshold(mut self, threshold: Duration) -> Self {
```
Comment (imabdulbasit) [inline src/data_source/storage/sql.rs:367]: do we want to set this at sequencer side through an env variable?

**Review by imabdulbasit — APPROVED** [review]

File: src/data_source/storage/sql.rs (hunk 2 above)
Comment (jbearer) [reply src/data_source/storage/sql.rs:367]: probably yeah

**Pull request merged**